lib/Importer/ONNXIFILoader.cpp

```diff
@@ -77,10 +76,9 @@ static bool loadWeight(const onnxTensorDescriptorV1 &in, Tensor *T) {
       TH.raw(i) = data[i];
     }
   } else if (in.dataType == ONNXIFI_DATATYPE_UINT64) {
-    // TODO: either switch IndexTy to be 64 bit, or switch to another type here.
-    T->reset(ElemKind::IndexTy, dims);
+    T->reset(ElemKind::Int64ITy, dims);
 
-    auto TH = T->getHandle<size_t>();
+    auto TH = T->getHandle<int64_t>();
```

Comment:
Question: `ONNXIFI` uses `ONNXIFI_DATATYPE_UINT64`, but we use a singed `int64_t` here. It is intentional? The value range of `int64_t` is smaller.

Reply:
Great catch! The next line here casts the data pointer into `int64_t`, and so I didn't notice it was based on `ONNXIFI_DATATYPE_UINT64`.

@yinghai @rdzhabarov Was this intentional? Perhaps this should be kept `uint64_t`, and then I can add an assert in the loop below here loading the tensor data to ensure we don't overflow `int64_t`? At least until we decide if we want to add `ElemKind::UInt64ITy`.

Reply:
Yes. I think uint64_t is fine for now. The only use case for int64_t so far is shape size which should be nonnegative.

Reply:
Done -- I added an assert here in the loop.